### PR TITLE
Go integ test for Operator chart and CI

### PR DESCRIPTION
**Pull request by levan-m**
#### What this PR does / why we need it:

Adds two integration tests which runs against Kind cluster in CI or locally. Test installs Operator helm chart, applies basic `DatadogAgent` manifest and asserts that Operator and Agent, DCA, CLC pods get to running status.
1. Vanilla install - install Operator from current charts with all defaults and *repo CRD*, and apply `v2alpha1` manifest. Then it asserts that specific number of pods reach running state.
2. Migration setup - installs cert manager and Operator with webhook enabled. Applies `v1alpha1` manifest and asserts specific number of pods reach running state. This test is skipped for K8S version <1.22, because of an issue with cert manager with older Kubernetes versions. This may be eventually fixed but not high priority.

Sample CI test [run](https://github.com/DataDog/helm-charts/actions/runs/5192774513/jobs/9362404194?pr=1027)

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [ ] `CHANGELOG.md` has been updated
- [ ] Variables are documented in the `README.md`

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -132,11 +128,45 @@ jobs:
...
+          node_image: kindest/node:${{ matrix.k8s }}
+          config: .github/kind_config.yaml
+      - name: Run integ tests
+        env:
+          LEVAN_M_TEST_API_KEY: ${{ secrets.LEVAN_M_TEST_API_KEY }}
+          LEVAN_M_TEST_APP_KEY: ${{ secrets.LEVAN_M_TEST_APP_KEY }}
+        run: |
+          kubectl cluster-info
+          kubectl get nodes
+          helm dependency build ./charts/datadog-operator
```
Comment (levan-m) [inline .github/workflows/ci.yaml:161]:
```suggestion
          helm repo add datadog https://helm.datadoghq.com
          helm dependency build ./charts/datadog-operator
```

File: test/integ/operator_integ_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+func verifyAgent(t *testing.T, kubectlOptions *k8s.KubectlOptions) {
+	verifyNumPodsForSelector(t, kubectlOptions, 2, "agent.datadoghq.com/component=agent")
+	verifyNumPodsForSelector(t, kubectlOptions, 1, "agent.datadoghq.com/component=cluster-agent")
+	verifyNumPodsForSelector(t, kubectlOptions, 1, "agent.datadoghq.com/component=cluster-checks-runner")
+}
+
+func verifyNumPodsForSelector(t *testing.T, kubectlOptions *k8s.KubectlOptions, numPods int, selector string) {
+	t.Log("Waiting for number of pods created", "number", numPods, "selector", selector)
+	k8s.WaitUntilNumPodsCreated(t, kubectlOptions, v1.ListOptions{
```
Comment (fanny-jiang) [inline test/integ/operator_integ_test.go:126]: this is pretty cool, waiting for the pods

**Review by fanny-jiang — CHANGES_REQUESTED** [review]
This looks great! It'd be awesome if we can update the contributing doc with instructions for running the integ tests locally. 

Not needed for this PR, but in the future maybe we can add test tooling to create the local kind cluster prior to running the tests.

File: test/integ/operator_integ_test.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	apiKeyEnv     = "API_KEY"
+	appKeyEnv     = "APP_KEY"
+	k8sVersionEnv = "K8S_VERSION"
```
[the remark this answers is not in the document]
Comment (levan-m) [reply test/integ/operator_integ_test.go:24]: Update doc

File: test/integ/operator_integ_test.go (hunk 2 above)
Comment (levan-m) [reply test/integ/operator_integ_test.go:128]: Agree, terratest provides really convenient interface!

**Comment by levan-m** [comment]
> Not needed for this PR, but in the future maybe we can add test tooling to create the local kind cluster prior to running the tests.

I like this idea, created a card in the backlog!

**Comment by buraizu** [comment]
Thanks, created DOCS-5748 to review

**Review by cswatt — APPROVED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -31,20 +40,43 @@ They may work with older versions, though.
...
- make test
+ make unit-test
  ```
 
+For changes which require baseline file update run `make update-test-baselines`. This will update all baseline files which should be included in the PR and pushed upstream.
+
+#### Integration Tests
+Integration tests run against locally configured context. We use [Terratest][terratest] for interacting with Helm and Kubectl.
+
+Each test creates unique namespace and subsequent resources are created in this namespace. Clean-up upon test completion is best effort, so it's recommended to run test against disposable cluster. **Make sure your don't accidentally run the test against production cluster.**
```
Comment (fanny-jiang) [inline CONTRIBUTING.md:54]:
```suggestion
Each test creates a unique namespace and subsequent resources are created in this namespace. Clean-up upon test completion is best effort, so it's recommended to run test against disposable cluster. **Make sure you don't accidentally run the test against a production cluster.**
```

**Review by fanny-jiang — APPROVED** [review]

**Pull request merged**